Blocstagram/BLCDataSource.m

```diff
@@ -349,4 +349,80 @@ - (NSString *) pathForFilename:(NSString *) filename {
 
 
 
+
+#pragma mark - Liking Media Items
+
+- (void) toggleLikeOnMediaItem:(BLCMedia *)mediaItem{
+    NSString *urlString = [NSString stringWithFormat:@"media/%@/likes", mediaItem.idNumber];
+    NSDictionary *parameters = @{@"access_token": self.accessToken};
+    
+    if (mediaItem.likeState == BLCLikeStateNotLiked) {
+        mediaItem.likeState = BLCLikeStateLiking;
+        
+        mediaItem.likeState = BLCLikeStateLiked; // faking as though got a positive response back
+        mediaItem.likesCount += 1; // faking as though got a positive response back
+
+//
+//        
+//        [self.instagramOperationManager POST:urlString parameters:parameters success:^(AFHTTPRequestOperation *operation, id responseObject) {
+//            mediaItem.likeState = BLCLikeStateLiked;
+//            [self reloadMediaItem:mediaItem];
+//        } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+//            
+//            //            mediaItem.likeState = BLCLikeStateNotLiked;
+//            [self reloadMediaItem:mediaItem];
+//            NSLog(@"failed to POST like to instagram");
+//        }];
+    } else if (mediaItem.likeState == BLCLikeStateLiked) {
+        
+        mediaItem.likeState = BLCLikeStateUnliking;
+        mediaItem.likeState = BLCLikeStateNotLiked; // faking as though got a positive response back
+        mediaItem.likesCount -= 1; // faking as though got a positive response back
+//
+//        
+//        
+//        
+//        [self.instagramOperationManager DELETE:urlString parameters:parameters success:^(AFHTTPRequestOperation *operation, id responseObject) {
+//            mediaItem.likeState = BLCLikeStateNotLiked;
+//            [self reloadMediaItem:mediaItem];
+//        } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+//            
+////            mediaItem.likeState = BLCLikeStateLiked;
+//            [self reloadMediaItem:mediaItem];
+//        }];
+//        
+    }
+    
+    [self reloadMediaItem:mediaItem]; // why are we calling this again - is this to cater for likeState = liking/unliking - if so why not just create an else statement and then put this in there - wouldn't that be more efficient?
```

Comment:
ok - I've uncommented it and pushed it github again

On Fri, Apr 24, 2015 at 1:28 PM, Falko Buttler <notifications@github.com>
wrote:

> In Blocstagram/BLCDataSource.m
> <https://github.com/rmtreks1/BlocStagram/pull/13#discussion_r29021700>:
>
> > +//
> > +//
> > +//
> > +//
> > +//        [self.instagramOperationManager DELETE:urlString parameters:parameters success:^(AFHTTPRequestOperation *operation, id responseObject) {
> > +//            mediaItem.likeState = BLCLikeStateNotLiked;
> > +//            [self reloadMediaItem:mediaItem];
> > +//        } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
> > +//
> > +////            mediaItem.likeState = BLCLikeStateLiked;
> > +//            [self reloadMediaItem:mediaItem];
> > +//        }];
> > +//
> > +    }
> > +
> > +    [self reloadMediaItem:mediaItem]; // why are we calling this again - is this to cater for likeState = liking/unliking - if so why not just create an else statement and then put this in there - wouldn't that be more efficient?
>
> This is used since it shows the spinner during liking/unliking
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rmtreks1/BlocStagram/pull/13/files#r29021700>.
>